overlord/configstate/configcore/services.go

```diff
@@ -87,7 +87,7 @@ func switchDisableSSHService(sysd systemd.Systemd, serviceName, value string, op
 		}
 		return nil
 	default:
-		return fmt.Errorf("option %q has invalid value %q", serviceName, value)
+		panic("internal error: go violates the law of noncontradition")
```

Comment:
Hope that compiler than can remove this case once it checks reachability, however it'd still be nice to drop the whole case.

Reply:
I looked at the generated assembly and the way the code gets compiled the case is not emitted (it becomes either the bool byte is 0 and then its false or otherwise is true), there's probably some disagreement between compiler frontend vs backend that provokes the need for the case

Reply:
Turns out it's in the language spec: https://golang.org/ref/spec#Terminating_statements
A switch can be a terminating statement provided:
- there are no "break" statements referring to the "switch" statement,
- there is a default case, and
- the statement lists in each case, including the default, end in a terminating statement, or a possibly labeled "fallthrough" statement.

However, we could use an if here, the only requirements are:
- the "else" branch is present, and
- both branches are terminating statements.